lib/travis/api/v3/queries/env_vars.rb

```diff
@@ -0,0 +1,27 @@
+module Travis::API::V3
+  class Queries::EnvVars < Query
+    params :id, :name, :value, :public, prefix: :env_var
+
+    def find(repository)
+      repository.env_vars    
+    end
+
+    def create(repository)
+      env_var = repository.env_vars.create(env_var_params)
+      handle_errors(env_var) unless env_var.valid?
+      repository.save!
+      env_var
+    end
+
+    private
+
+      def handle_errors(env_var)
+        base = env_var.errors[:base]
+        case
```

Comment:
does this need to be a case statement? This is totally a style/readability thing, but I would do:
``` ruby
raise WrongParams if base.include?(:format) 
raise DuplicateResource if base.include?(:duplicate_resource)
raise ServerError
```